ethstorage/miner/l1_mining_api.go

```diff
@@ -42,6 +42,34 @@ type l1MiningAPI struct {
 	lg log.Logger
 }
 
+// Pre-check if the miner has been whitelisted before actually mining
+func (m *l1MiningAPI) CheckMinerRole(ctx context.Context, contract, miner common.Address) error {
+	enforced, err := m.PollingClient.ReadContractField("enforceMinerRole", nil)
+	if err != nil {
+		return fmt.Errorf("failed to query enforceMinerRole(): %w", err)
```

Comment:
The new version of the ES Node should remain compatible with the existing storage contract.